livre/core/render/FrameInfo.h

```diff
@@ -31,10 +31,12 @@ namespace livre
 struct FrameInfo
 {
     LIVRECORE_API FrameInfo( const Frustum& frustum,
-                             const uint32_t frameId );
+                             const uint32_t frameId,
+                             const uint32_t incarnationId );
 
     Frustum frustum; //!< The current frustum.
     uint32_t frameId ; //!< The current frame id.
+    uint32_t incarnationId; //!< The incarnation for each rendered frame
```

Comment:
then I would suggest renaming this value to frameId. I'm not sure incarnation has the correct meaning :-) http://www.dictionary.com/browse/incarnation